### ISSUE TITLE
Design Intent - Accessibility Feedback - Digital GI Bill, GI Bill Comparison Tool: Comparison Tool Enhancements

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#45139]
## VFS acceptance criteria
- [ ] Let Platform know when the **Must** feedback has been incorporated 
- [ ] Leave any comments for feedback you decide _not_ to take
- [ ] VFS team closes the ticket

## Thoughts/questions
- General feedback: The GI Bill Comparison Tool has a lot of moving parts and a lot of things that could go wrong from an accessibility perspective. I think the direction you're going from the existing tool is the right direction, but definitely make sure you budget time for thinking about and testing for accessibility.
- Angela Fowler's open accessibility office hours were mentioned in the meeting, and I'll plug them here too: Fridays at 11:30am Eastern Time, with a Zoom link posted each Friday in the #vagov-accessibility channel in DSVA Slack. She's a native screen reader user who will have a very helpful perspective, and I'd even encourage you to ask her to walk through some pieces of the existing tool to highlight where some barriers might be.
- I've noted a couple of specific items I see in your designs as "Must" feedback, and I've provided some broader recommendations in the "Should" feedback. Note that the "Should" items aren't less important! They're just broad enough that there are many possible approaches for ensuring accessibility and putting them under "Must" didn't feel like a good fit.
- Please let me know if you have any questions, or if there's anything you want to talk through. I'm happy to meet during [Collaboration Cycle office hours](https://depo-platform-documentation.scrollhelp.site/support/office-hours) or in a separate scheduled meeting.

## Feedback
- Practice areas will document their feedback following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback).

**Must:**
- For sticky content (especially the comparison drawer), be careful about high-zoom situations. The VA.gov experience standards and WCAG 2.1 AA require [support for up to roughly 400% zoom](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html) without loss of functionality or requiring horizontal scrolling (vertical scrolling okay). Sticky content often doesn't play well with high zoom levels, covering up some/all information on the page. You'll want to select an appropriate break point where the sticky controls become not-sticky.
- Make sure links and buttons have unique accessible names (either the link/button text itself is unique, or there's an `aria-label` attribute that can be added to the link/button to provide screen reader users a unique version). This includes things like the "Learn more" additional info components in the filters. Two examples of where this can be an issue:
   - When a screen reader user is scanning the page, they often have their screen reader announce a list of the links and buttons, and skip straight to the link/button that sounds like it might be going where they want to go. Without the surrounding context of the page, it's not possible to distinguish between links/buttons with the same text.
   - When a voice command user tells their computer "Click [link text]," their software attempts to match whatever they said to the links or buttons on the page. If there's more than one, it can't make a match and requires the user to take extra steps to narrow down from there. 

**Should:**
- Now is a good time to start thinking about organizing page content by heading -- even if some of those headings aren't visible. Screen reader users frequently scan the page looking for what they need by having their screen reader announce the heading hierarchy: H1 = page title, H2s = major sections, H3s = sub-sections, etc. Good heading structure can make a huge difference for page navigability. In situations where headings aren't visually necessary, they can be hidden via CSS but still announced by screen readers. For each page of the comparison tool, I would recommend making an outline (think 8th grade English class) to break it out into sections and subsections, then assigning an appropriate heading. If they work visually, great! If not, make a note in your designs for a hidden heading for when it goes to code.
- The VA.gov experience standards and WCAG 2.1 AA require [support for up to roughly 400% zoom](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html) without loss of functionality or requiring horizontal scrolling (vertical scrolling okay). When you have something as information-dense as the comparison tool, this can be a huge challenge -- sort of an extreme version of mobile/responsive design. I would recommend working on some high-zoom prototypes and start thinking about possible interaction patterns. Example: If the results map takes up the whole viewport at high zoom, what should happen when someone clicks a map marker? The more you plan for this early on the easier it will be to implement. 
- Now is also a good time to start thinking about keyboard navigation. Keyboard-only users frequently navigate the page by pressing [Tab] or [Shift] + [Tab] to move between interactive elements (links, buttons, form fields). With large amounts of content like what's on the comparison tool, it's important to be deliberate about where elements fall in the tab order. Example: In the current comparison tool, if I search my zip code and no filters, within the default 25 miles I get 461 results. If I select the "Compare" checkbox for the first three of them and the compare drawer opens at the bottom, I have to press [Tab] 1,401 times to get to the "Compare" button in the drawer to open the Institution Comparison view (including needing to tab through all 461 map markers). As you finalize your designs for this iteration, it's worth annotating the tab stops and drawing in the tab order path, like a zig-zag line through the page to show where keyboard focus goes. If the line doesn't make sense or important controls are too hard to get to, reconsider where they're placed. 

**Consider:**

## Platform directions
- Remove the governance-team label
- Update "Issue Title"
- Link to collab cycle Request epic
- Add your feedback
- Add assignees based on collab cycle touchpoint
  - **Design Intent**: VFS designer, VFS PM (optional), yourself (optional)
  - **Midpoint Review**: VFS PM, yourself (optional)
  - **Staging Review QA Only**: VFS PM, yourself (optional)
